holoviews/plotting/bokeh/element.py

```diff
@@ -2421,17 +2428,34 @@ def _draw_scalebar(self, plot):
         else:
             base_unit = unit
 
-        _default_scalebar_opts = {"background_fill_alpha": 0.8}
-        opts = dict(_default_scalebar_opts, **self.scalebar_opts)
+        if self._subcoord_overlaid:
+            srange = plot.renderers[-1].coordinates.y_source
```

Comment:
```suggestion
            srange = renderer.coordinates.y_source
```

Is plot.renderers order guaranteed? It looks like the calling function `_init_glyphs` already gets the correct renderer so maybe it would be more robust to just pass that in to _draw_scalebar?

https://github.com/holoviz/holoviews/blob/f9c7f05c4078cc7ebb571c6b5aa10c1cc2662962/holoviews/plotting/bokeh/element.py#L2113-L2114

```python
            self._draw_scalebar(plot, renderer)
```